### PR TITLE
Limit switch

### PR DESCRIPTION
**Pull request by JonathanLindsey**
I first fixed the typo in the navigation menu: Limit Switch had been spelled "Limi Switch".

Then I improved the Limit Switch tutorial with documentation on the Normally Closed input to the Limit Switch Open VI, explaining it is similar to the Invert input on the Motor Open VI.

### CODE REVIEW

**Comment by JonathanLindsey** [comment]
Please check that I did the VI snippet correctly. I noticed when I'm serving the website locally with `jekyll serve` I cannot drag and drop the VI snippets directly of the webpage (from FireFox) to LV whereas I can from the real website.

**Comment by mshafer1** [comment]
Thanks for contributing! I'll check the snippet when I get home tonight.

**Comment by JonathanLindsey** [comment]
I think I need to change the screenshot I added of context help from a downloadable image to a zoomable image. I will do that later today. What is the lgImgPanel.html for?

**Comment by mshafer1** [comment]
lgImgPanel is the full screen preview that appears when a zoomableimage is clicked on.

I am able to drag your new open vi snippet into LV on my machine (and have it work).

File: _tutorials/sensors/roborio/limit-switch/index.md [hunk 1]
```diff
@@ -17,7 +17,11 @@ For the sake of this tutorial, we will construct a VI that takes in the DevRef's
 
 Just a formality, but we will start by opening the necessary hardware in Begin.vi. For this example, we will use two limit switches, a joystick, and a motor.
 
-{% include downloadableImage.html path='images/LimitSwitch_Begin.png' alt='Opening the limit switch in begin.vi' height=507 width=220 %}
+{% include downloadableImage.html path='images/LimitSwitch_Begin.png' alt='Opening the limit switch in begin.vi' height=278 width=653 %}
+
+Notice the Normally Closed false constants connected to the Limit Switch Open VIs. They work similarly to the Invert constant on the Motor Open VI. If you find the switch reports true when you expect false and false when you expect true, change the value of the Normally Closed input for that switch.
+
+{% include downloadableImage.html path='images/LimitSwitchOpen_ContextHelp.png' alt='Limit switch open Context Help' height=373 width=257 %}
```
Comment (mshafer1) [inline _tutorials/sensors/roborio/limit-switch/index.md]: Yes, this should be 
```
{% include zoomableImage.html path='images/LimitSwitchOpen_ContextHelp.png' alt='Limit switch open Context Help' height=373 width=257 %}
```

the zoomableImage template ([source](https://github.com/mshafer1/frclabviewtutorials/blob/master/_includes/zoomableImage.html)) creates an image with an "onclick=showImage" registered. That method uses the #largeImgPanel div to display a full screen view of the image.

File: _tutorials/sensors/roborio/limit-switch/index.md [hunk 2]
```diff
@@ -39,7 +43,7 @@ Now, this is good, but we wanted the limit switches to turn off the motor at its
 {% include downloadableImage.html path='images/LimitSwitch_BasicWLimits.png' alt='Basic use of limit switch' height=321 width=650 %}
 
 
-**NOTE: This is configured for the limit switch to report true when pressed, depending on how it is wired, this may be backwards, if this is the case, merely insert a not gate in between the read and the select.**
+**NOTE: If the limit switch reports true when you expect false and false when you expect true, either insert a Not function between the Read and the Select or change the value of the Normally Closed input to the Limit Switch Open VI.**
```
Comment (mshafer1) [inline _tutorials/sensors/roborio/limit-switch/index.md]: If the reader hasn't wired the switch yet, how can we guide them to make the correct choice for how to wire it for their application?

File: _tutorials/sensors/roborio/limit-switch/index.md (hunk 1 above)
Comment (JonathanLindsey) [reply _tutorials/sensors/roborio/limit-switch/index.md]: While we're here, what is the 'alt' text (i.e. where does it show up in the rendered webpage? Is it so that the image is searchable and/or if for some reason the reader has image rendering turned off?

On the [sensors/dashboard/gyro page](https://github.com/mshafer1/frclabviewtutorials/blob/master/_tutorials/sensors/dashboard/gyro/index.md) (which I see has some issues with some images) the first image is zoomable and has `scaler=''` in the image line. Does that mean anything? I tried changing the value of scaler to a few different values but nothing changed.

How do I change the size that zoomable images display? Here I've set the height and width to the same values but the zoomable one shows up much smaller.
image
```
{% include downloadableImage.html path='images/LimitSwitchOpen_ContextHelp.png' alt='Limit switch open Context Help' height=373 width=257 %}
{% include zoomableImage.html path='images/LimitSwitchOpen_ContextHelp.png' alt='Limit switch open Context Help' height=373 width=257 %}
```

File: _tutorials/sensors/roborio/limit-switch/index.md (hunk 2 above)
Comment (JonathanLindsey) [reply _tutorials/sensors/roborio/limit-switch/index.md]: I'll add this to the note.

If (a) you wire the switch with the roboRIO's GND and Sig wires (like you should) (b) the wires connect to the GND and Normally Open (NO) terminals on the switch (regardless of which wire goes to which terminal) and (c) the Switch Open VI has a False constant wired to its Normally Closed input (or is not wired at all), the Switch Get VI will output True when the switch is pressed (closed) and False when the switch is not pressed (open).

**Comment by JonathanLindsey** [comment]
I found I have to have the server running in order to be able to pull the VI snippets from the webpage to LV (that makes since).

File: _tutorials/sensors/roborio/limit-switch/index.md (hunk 1 above)
Comment (mshafer1) [reply _tutorials/sensors/roborio/limit-switch/index.md]: alt is a parameter for alternate text - Firefox is the only browser that requires it as far as I know, but it is also good to provide for either the bad link or screen reader cases.

scaler is an artifact of something I tried previously - since the zoomable images don't need to be full size (as they can zoom in) and are often screen shots, I attempted at one point to create a scale factor to be applied to the sizing

It looks like it's currently hard coded as 2/3:
https://github.com/mshafer1/frclabviewtutorials/blob/master/_includes/zoomableImage.html
(although, know that I've learned a lot more about Jekyll since I did this conversion - I think making the scaler work would just require plugging it in as `times: include.scaler`)

I'll mess with this real quick and see if I can make it work a bit better.

File: _tutorials/sensors/roborio/limit-switch/index.md (hunk 2 above)
Comment (mshafer1) [reply _tutorials/sensors/roborio/limit-switch/index.md]: I find this hard to parse . . .

File: _tutorials/sensors/roborio/limit-switch/index.md [hunk 3]
```diff
@@ -38,21 +49,22 @@ Now, this is good, but we wanted the limit switches to turn off the motor at its
 
 {% include downloadableImage.html path='images/LimitSwitch_BasicWLimits.png' alt='Basic use of limit switch' height=321 width=650 %}
 
+*NOTE: If the Limit Switch Get VI reports true when you expect false and false when you expect true, you have a few options:*
 
-**NOTE: This is configured for the limit switch to report true when pressed, depending on how it is wired, this may be backwards, if this is the case, merely insert a not gate in between the read and the select.**
+1. Change your choice of wiring the NO or NC terminal on the physical switch.
```
Comment (mshafer1) [inline _tutorials/sensors/roborio/limit-switch/index.md]: Just a heuristic question, numbering implies rank. Is that intentional here? If so, what is the ordering?

Comment (JonathanLindsey) [reply _tutorials/sensors/roborio/limit-switch/index.md]: The order is electrical to programming. In the next commit I added a recomendation which I meant to do before.

**Comment by JonathanLindsey** [comment]
It has only been 20 minutes since I did the squash and merge into master.
I see the website has already been updated. However, the wiring image is not showing up on the real website whereas it had when served locally. The image of context help _is_ showing correctly.
Why would this be?

This might be helpful. When I did the squash and merge I got the following email. I got the same email a few minutes later when I squashed and merged PR #7 . This email may be totally irrelevant.

> The page build completed successfully, but returned the following warning for the `master` branch:
> 
> It looks like you're using GitHub Pages to distribute binary files. We strongly suggest that you use releases to ship projects on GitHub. Releases are GitHub's way of packaging and providing software to your users. You can think of it as a replacement to using downloads to provide software. We found the following file(s) which may be a good candidate for releases: _tutorials/filedAuto/files/Tool project/WinSCP/WinSCP-5.11.3-Portable/WinSCP.exe. For more information, see https://help.github.com/en/articles/about-releases.
> 
> For information on troubleshooting Jekyll see:
> 
>   https://help.github.com/articles/troubleshooting-jekyll-builds
> 
> If you have any questions you can contact us by replying to this email.

**Comment by mshafer1** [comment]
> This might be helpful. When I did the squash and merge I got the following email. I got the same email a few minutes later when I squashed and merged PR #7 . This email may be totally irrelevant.

Yep, unrelated - it's complaining that we have a zip of WinSCP on the site.

> It has only been 20 minutes since I did the squash and merge into master.
> I see the website has already been updated. However, the wiring image is not showing up on the real website whereas it had when served locally. The image of context help _is_ showing correctly.
> Why would this be?

Windows paths are case **in**sensitive, but gh-pages is served from Linux where paths are case **sensitive**. Looking at [index.md](https://github.com/mshafer1/frclabviewtutorials/blob/master/_tutorials/sensors/roborio/limit-switch/index.md), the path is .png, but looking at the [folder](https://github.com/mshafer1/frclabviewtutorials/tree/master/_tutorials/sensors/roborio/limit-switch/images), the path is .PNG.
